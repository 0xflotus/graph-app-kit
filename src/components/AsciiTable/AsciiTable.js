import React, { Component } from "react";
import * as PropTypes from "prop-types";
import asciitable from "ascii-data-table";

export class AsciiTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      serializedRows: []
    };
  }
  componentDidMount() {
    this.makeState(this.props);
  }
  componentWillReceiveProps(props) {
    this.makeState(props);
  }
  makeState(props) {
    const { data } = props;
    const serializedRows = asciitable.serializeData(data) || [];
    this.setState({ serializedRows });
  }
  render() {
    const { maxColWidth = 70 } = this.props;
    const { serializedRows } = this.state;
    if (!serializedRows.length) return null;
    return (
      <pre>
        {asciitable.tableFromSerializedData(serializedRows, maxColWidth)}
      </pre>
    );
  }
}

AsciiTable.propTypes = {
  data: PropTypes.array,
  maxColWidth: PropTypes.number
};

export default AsciiTable;
