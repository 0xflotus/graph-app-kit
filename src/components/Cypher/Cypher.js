import { Component } from "react";
import * as PropTypes from "prop-types";

export const missingRenderPropError =
  "Pass render property to render something. <Cypher render={({pending, error, response} => {})} />";
export const missingQueryError =
  "A property named 'query' is missing. Component needs a query to be executed.";
export const missingDriverError =
  "A property or conext object named 'driver' is missing. Component needs a connected driver to send queries over. See <Provider>.";

export class Cypher extends Component {
  constructor(props, context) {
    super(props, context);
    this.driver = props.driver || context.driver;
  }
  state = {
    result: null,
    pending: true,
    error: null,
    tick: 0
  };
  componentDidMount() {
    this.setupQuery(this.props);
  }
  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.pending !== nextState.pending) return true;
    if (this.state.tick !== nextState.tick) return true;
    if (nextProps.cTag && this.props.cTag !== nextProps.cTag) return true;
    return false;
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.cTag && this.props.cTag !== nextProps.cTag) {
      clearInterval(this.interval);
      this.setupQuery(nextProps);
    }
  }
  componentWillUnmount() {
    clearInterval(this.interval);
  }
  setupQuery(props) {
    this.query(props);
    if (props.interval) {
      this.interval = setInterval(
        () => this.query(props),
        props.interval * 1000
      );
    }
  }
  query(props) {
    const { query, params = undefined } = props;
    if (!query) throw new Error(missingQueryError);
    if (!this.driver) throw new Error(missingDriverError);
    this.setState({ pending: true }, () => {
      const session = this.driver.session();
      session
        .run(query, params)
        .then(res => {
          session.close();
          this.setState(state => ({
            pending: false,
            result: res,
            error: null,
            tick: state.tick + 1
          }));
        })
        .catch(error => {
          this.setState(state => ({
            pending: false,
            result: null,
            error: error,
            tick: state.tick + 1
          }));
        });
    });
  }
  render() {
    if (!this.props.render) {
      throw new Error(missingRenderPropError);
    }
    return this.props.render({ ...this.state });
  }
}

Cypher.contextTypes = {
  driver: PropTypes.object
};

Cypher.propTypes = {
  driver: PropTypes.object,
  params: PropTypes.object,
  query: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
  cTag: PropTypes.any
};

export default Cypher;
