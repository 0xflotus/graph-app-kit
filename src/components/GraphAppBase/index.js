export { GraphAppBase, CONNECTED, DISCONNECTED } from "./GraphAppBase";
export { ConnectModal } from "./ConnectModal";
