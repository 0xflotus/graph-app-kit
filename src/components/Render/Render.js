const Render = ({ if: cond, children }) => {
  return cond ? children : null;
};
export default Render;
