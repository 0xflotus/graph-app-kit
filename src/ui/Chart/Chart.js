import React from "react";
import CypherChartWrapper from "./wrappers/CypherChartWrapper";
import ChartSwitch from "./wrappers/ChartSwitch";

export const Chart = props => {
  return <div>Hello</div>;
};
