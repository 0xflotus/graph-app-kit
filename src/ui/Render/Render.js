export const Render = ({ if: cond, children }) => {
  return cond ? children : null;
};
