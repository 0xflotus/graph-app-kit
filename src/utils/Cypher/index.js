export {
  Cypher,
  missingRenderPropError,
  missingQueryError,
  missingDriverError
} from "./Cypher";
