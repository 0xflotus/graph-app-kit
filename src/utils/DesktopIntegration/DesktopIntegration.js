import { Component } from "react";
import { getActiveGraph, getCredentials, eventToHandler } from "./helpers";

export default class DesktopIntegration extends Component {
  setupListener() {
    const { integrationPoint } = this.props;
    if (integrationPoint && integrationPoint.onContextUpdate) {
      integrationPoint.onContextUpdate((event, newContext, oldContext) => {
        const handlerPropName = eventToHandler(event.type);
        if (!handlerPropName) return;
        if (typeof this.props[handlerPropName] === "undefined") return;
        this.props[handlerPropName](event, newContext, oldContext);
      });
    }
  }
  loadInitialContext() {
    const { integrationPoint, onMount = null } = this.props;
    if (integrationPoint && integrationPoint.getContext) {
      integrationPoint
        .getContext()
        .then(context => {
          const activeGraph = getActiveGraph(context);
          if (onMount) {
            const connectionCredentials = getCredentials(
              "bolt",
              activeGraph.connection
            );
            onMount(activeGraph, connectionCredentials, context);
          }
        })
        .catch(e => {}); // Catch but don't bother
    }
  }
  componentDidMount() {
    this.loadInitialContext();
    this.setupListener();
  }
  render() {
    return null;
  }
}
