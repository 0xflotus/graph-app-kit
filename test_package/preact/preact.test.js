import { h } from "preact";
/** @jsx h */
import { deep } from "preact-render-spy";

// Package exports ui/
import { Render } from "../../dist/ui/Render";
import { AsciiTable } from "../../dist/ui/AsciiTable";
import { Chart } from "../../dist/ui/Chart";
import { CodeMirror } from "../../dist/ui/Editor";

// Package exports utils/
import { Cypher } from "../../dist/utils/Cypher";
import { DesktopIntegration } from "../../dist/utils/DesktopIntegration";
import { DriverProvider } from "../../dist/utils/DriverProvider";
import { GraphAppBase } from "../../dist/utils/GraphAppBase";

// ui/
test("Render works", () => {
  const context = deep(
    <Render if={true}>
      <p>Hello</p>
    </Render>
  );
  expect(context.output()).toMatchSnapshot();
});
test("AsciiTable works", () => {
  const data = [["x"], ["y"]];
  const context = deep(<AsciiTable data={data} />);
  context.rerender();
  expect(context.text()).toMatchSnapshot();
});
test("Chart works", () => {
  const data = [{ label: "Used", value: 30 }, { label: "Free", value: 20 }];
  const context = deep(
    <Chart
      data={data}
      title="Static circular data"
      chartType="doughnut"
      type="json"
    />
  );
  expect(context.output()).toMatchSnapshot();
});
test("CodeMirror works", () => {
  const context = deep(<CodeMirror />);
  expect(context.output()).toMatchSnapshot();
});

// utils/
test("Cypher works", () => {
  const context = deep(
    <Cypher
      driver={resolvingDriver(0, "no!")}
      query="RETURN rand() as n"
      render={({ pending, error, result, tick }) => {
        return pending ? "pending" : error ? error.message : result;
      }}
    />
  );
  expect(context.output()).toMatchSnapshot();
});
test("DesktopIntegration works", () => {
  const context = deep(<DesktopIntegration integrationPoint={true} />);
  expect(context.output()).toMatchSnapshot();
});
test("DriverProvider works", () => {
  const context = deep(
    <DriverProvider driver={resolvingDriver(0, "yes")}>Hello</DriverProvider>
  );
  expect(context.output()).toMatchSnapshot();
});
test("GraphAppBase works", () => {
  const context = deep(
    <GraphAppBase
      integrationPoint={null}
      render={() => <p>Hello</p>}
      driverFactory={{ driver: () => resolvingDriver(0, "yes") }}
    />
  );
  expect(context.output()).toMatchSnapshot();
});

// Helpers
const sleep = (secs, shouldResolve = true, message) =>
  new Promise((resolve, reject) => {
    setTimeout(
      () => (shouldResolve ? resolve(message) : reject(new Error(message))),
      secs * 1000
    );
  });

const driver = (shouldResolve = true, waitSecs = 0, message = "") => ({
  session: () => ({
    close: () => {},
    run: () => sleep(waitSecs, shouldResolve, message)
  })
});

const resolvingDriver = (waitSecs = 0, resolveTo = "Resolved") =>
  driver(true, waitSecs, resolveTo);
