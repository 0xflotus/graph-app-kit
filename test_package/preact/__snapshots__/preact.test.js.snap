// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsciiTable works 1`] = `
"╒═══╕
│\\"x\\"│
╞═══╡
│\\"y\\"│
└───┘"
`;

exports[`Chart works 1`] = `
<div>
  <div>
    <undefined></undefined>
    <div
      style="display: flex; flex-direction: column;"
      class="vx-legend"
    >
    </div>
  </div>
  <h2>Static circular data</h2>
</div>
`;

exports[`Cypher works 1`] = `"pending"`;

exports[`CypherEditor works 1`] = `<div class="ReactCodeMirror"></div>`;

exports[`DesktopIntegration works 1`] = `null`;

exports[`DriverProvider works 1`] = `"Hello"`;

exports[`GraphAppBase works 1`] = `
<div>
  <p>Hello</p>
</div>
`;

exports[`Render works 1`] = `<p>Hello</p>`;

exports[`Sidebar works 1`] = `
<div>
  <div
    style="min-height: 300px; max-height: auto;"
    class="pushable"
  >
    <div
      style="width: auto;"
      class="ui vertical ui push left visible sidebar menu"
    >
      <div class="pusher">
        <div class="ui basic segment"></div>
      </div>
    </div>
  </div>
  <a
    onClick={[Function anonymous]}
    class="item"
  >
  </a>
</div>
`;
